tests/test_raster/test_interpolate.py

```diff
@@ -0,0 +1,568 @@
+        j = np.random.default_rng(42).integers(1, 49, size=10)
+        x, y = r.ij2xy(i, j)
+        vals = r.interp_points((x, y), method=method, force_scipy_function="map_coordinates")
+        vals2 = r.interp_points((x, y), method=method, force_scipy_function="interpn")
+
+        assert np.array_equal(vals, vals2, equal_nan=True)
+
+        # 3/ Test return_interpolator is consistent with above
+        interp = _interp_points(
+            r.get_nanarray(),
+            transform=r.transform,
+            area_or_point=r.area_or_point,
+            points=(x, y),
+            method=method,
+            return_interpolator=True,
+        )
+        vals3 = interp((y, x))
+        vals3 = np.array(np.atleast_1d(vals3), dtype=np.float32)
+
+        assert np.array_equal(vals2, vals3, equal_nan=True)
+
+    @pytest.mark.parametrize("example", [landsat_b4_path, aster_dem_path])  # type: ignore
+    @pytest.mark.parametrize(
+        "method", ["nearest", "linear", "cubic", "quintic", "slinear", "pchip", "splinef2d"]
+    )  # type: ignore
+    @pytest.mark.parametrize("dist", ["half_order_up", "half_order_down", 0, 1, 5])  # type: ignore
+    def test_interp_point__nodata_propag(
+        self,
+        example: str,
+        method: Literal["nearest", "linear", "cubic", "quintic", "slinear", "pchip", "splinef2d"],
+        dist: Literal["half_order_up", "half_order_down"] | int,
+    ) -> None:
+
+        # Open and crop for speed
+        r = gu.Raster(example)
+        r = r.crop((r.bounds.left, r.bounds.bottom, r.bounds.left + r.res[0] * 100, r.bounds.bottom + r.res[1] * 100))
+
+        # 1/ Check the propagation of NaNs
+
+        # 1.1/ Manual check for a specific point
+        # Convert raster to float
+        r = r.astype(np.float32)
+
+        # Create a NaN at a given pixel (we know the landsat example has no NaNs to begin with)
+        i0, j0 = (10, 10)
+        r[i0, j0] = np.nan
+
+        # Create a big NaN area in the bottom right (for more complex NaN propagation below)
+        r[80:90, 80:90] = np.nan
+
+        # All surrounding pixels with distance half the method order rounded up should be NaNs
+        order = method_to_order[method]
+        d = _dist_nodata_spread(order=order, dist_nodata_spread=dist)
+
+        # Get indices of NaNs within the distance from NaNs
+        indices_nan = [
+            (i0 + i, j0 + j) for i in np.arange(-d, d + 1) for j in np.arange(-d, d + 1) if (np.abs(i) + np.abs(j)) <= d
+        ]
+        i, j = list(zip(*indices_nan))
+        x, y = r.ij2xy(i, j)
+        vals = r.interp_points((x, y), method=method, force_scipy_function="map_coordinates", dist_nodata_spread=dist)
+        vals2 = r.interp_points((x, y), method=method, force_scipy_function="interpn", dist_nodata_spread=dist)
+
+        assert all(np.isnan(np.atleast_1d(vals))) and all(np.isnan(np.atleast_1d(vals2)))
+
+        # Same check for values within one pixel of the exact coordinates
```

Comment:
Is this not already included in the previous test?

Reply:
I added a comment to clarify: previous test checked for points falling right on grid coordinates, this one is for points falling anywhere within half a pixel of the previous grid coordinates.